### PR TITLE
Fix #590: Full UI Switch Profile Option

### LINKED ISSUE #590: Full UI: Switch Profile Option in Navigation Drawer
The Navigation Drawer needs `Switch Profile` option. (The svg file for the icon will be shared on email.)
Screenshot 2020-01-06 at 2 58 29 PM

When learner clicks on this option, it should open the alert dialog which is shown when learner presses back button on home screen.
Screenshot 2020-01-06 at 2 58 39 PM

Also, add test cases to test this functionality.

## Checklist
- [ ] Change the text of `home_activity_back_dialog_message` in strings.xml file as per mocks.
- [ ] Add `Switch Profile` option along with its functionality.
- [ ] Add test cases for the `Switch Profile` options. (Check `NavigationDrawerTestActivity` and `NavigationDrawerTestActivityTest` files for writing test cases)

### PR DESCRIPTION
**Pull request by PrarabdhGarg**
## Explanation

Fixes #590 
This pull requests adds the switch profile option in the navigation drawer, along with the UI tests associated with it

## Checklist

- [ x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x ] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [ x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x ] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [ x] The PR is made from a branch that's **not** called "develop".
- [x ] The PR is made from a branch that is up-to-date with "develop".
- [x ] The PR's branch is based on "develop" and not on any other branch.
- [ x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by PrarabdhGarg** [comment]
@rt4914 I think this pull request also contains commits that were made when I updated my develop branch. How can I avoid these commits from being a part of my pull request?

**Comment by rt4914** [comment]
> @rt4914 I think this pull request also contains commits that were made when I updated my develop branch. How can I avoid these commits from being a part of my pull request?

Actually your branch is 4 commits behind oppia-develop. Because of which you are getting this unnecessary code and also your test cases are failing. I suggest you merge your branch with latest oppia-develop code. It should fix the issue if done correctly.

`git checkout fix-590-full-ui-switch-profile-option`
`git fetch upstream`
`git merge upstream/develop`
If there are any merge conflicts after this step (which they are in NavigationDrawerFragmentPresenter and NavigationDrawerTestActivityTest ), resolve those merge conflicts.
Then add / commit / push your code.

File: app/src/main/java/org/oppia/app/drawer/NavigationDrawerFragmentPresenter.kt [hunk 1]
```diff
@@ -84,16 +86,31 @@ class NavigationDrawerFragmentPresenter @Inject constructor(
...
+          AlertDialog.Builder(fragment.context!!, R.style.AlertDialogTheme)
+            .setMessage(R.string.home_activity_back_dialog_message)
+            .setNegativeButton(R.string.home_activity_back_dialog_cancel) { dialog, _ ->
+              dialog.dismiss()
+            }
+            .setPositiveButton(R.string.home_activity_back_dialog_exit) { _, _ ->
+              // TODO(#322): Need to start intent for ProfileActivity to get update. Change to finish when live data bug is fixed.
+              val intent = Intent(fragment.activity, ProfileActivity::class.java)
+              intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
+              fragment.activity!!.startActivity(intent)
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/drawer/NavigationDrawerFragmentPresenter.kt:109]: Prefer not to call intent this way. You might have found this in some Profile related code but that will be updated with time.
Instead prefer calling intend similar to HomeActivity code. Mainly you will need to create a companion object for intent in the ProfileActivity and then you can call that function wherever you want.

File: app/src/sharedTest/java/org/oppia/app/testing/NavigationDrawerTestActivityTest.kt [hunk 2]
```diff
@@ -147,6 +151,26 @@ class NavigationDrawerTestActivityTest {
     }
   }
 
+  @Test
+  fun testNavigationDrawerTestActivity_openNavigationDrawer_selectSwitchProfileMenu_showsExitToProfileChooserDialog() {
+    launch(NavigationDrawerTestActivity::class.java).use {
+      onView(withId(R.id.home_activity_drawer_layout)).perform(open())
+      onView(withText(R.string.menu_switch_profile)).perform(click())
+      onView(withText(R.string.home_activity_back_dialog_message)).check(matches(ViewMatchers.isDisplayed()))
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/app/testing/NavigationDrawerTestActivityTest.kt]: Update: For better code clarity write it like this:
`onView(withText(R.string.home_activity_back_dialog_message)).check(matches(isDisplayed()))`

Ditto in below test case.

Comment (rt4914) [reply app/src/sharedTest/java/org/oppia/app/testing/NavigationDrawerTestActivityTest.kt]: Also, after this change make sure to check that you do not have any unused imports.

File: app/src/sharedTest/java/org/oppia/app/testing/NavigationDrawerTestActivityTest.kt [hunk 3]
```diff
@@ -147,6 +151,26 @@ class NavigationDrawerTestActivityTest {
...
+  }
+
+  @Test
+  fun testNavigationDrawerTestActivity_openNavigationDrawer_selectSwitchProfileMenu_showsExitToProfileChooserDialog_clickExit_checkOpensProfileActivity() {
+    launch(NavigationDrawerTestActivity::class.java).use {
+      onView(withId(R.id.home_activity_drawer_layout)).perform(open())
+      onView(withText(R.string.menu_switch_profile)).perform(click())
+      onView(withText(R.string.home_activity_back_dialog_message)).check(matches(ViewMatchers.isDisplayed()))
+      onView(withText(R.string.home_activity_back_dialog_exit)).perform(click())
+      intended(IntentMatchers.hasComponent(ProfileActivity::class.java.name))
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/app/testing/NavigationDrawerTestActivityTest.kt]: Update:

`intended(hasComponent(ProfileActivity::class.java.name))`

**Review by rt4914 — CHANGES_REQUESTED** [review]
Great. This looks really clean. I have just requested some few changes. PTAL

**Comment by PrarabdhGarg** [comment]
@rt4914 I have made the required changes

**Review by rt4914 — APPROVED** [review]
LGTM, Thanks.

**Comment by rt4914** [comment]
@PrarabdhGarg before merging this PR, update your PR title to

`Fix #590: Full UI Switch Profile Option`

**Comment by PrarabdhGarg** [comment]
> @PrarabdhGarg before merging this PR, update your PR title to
> 
> `Fix #590: Full UI Switch Profile Option`

@rt4914 Done

**Comment by rt4914** [comment]
> > @PrarabdhGarg before merging this PR, update your PR title to
> > `Fix #590: Full UI Switch Profile Option`
> 
> @rt4914 Done

Actually you missed out `#`, this is important as it automatically closes the corresponding issue.

**Pull request merged**